op-deployer/pkg/deployer/bootstrap/implementations.go

```diff
@@ -0,0 +1,207 @@
+package bootstrap
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"errors"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum-optimism/optimism/op-deployer/pkg/deployer/artifacts"
+	"github.com/ethereum-optimism/optimism/op-deployer/pkg/deployer/broadcaster"
+	"github.com/ethereum-optimism/optimism/op-deployer/pkg/deployer/opcm"
+	"github.com/ethereum-optimism/optimism/op-deployer/pkg/env"
+	"github.com/ethereum-optimism/optimism/op-service/cliutil"
+	opcrypto "github.com/ethereum-optimism/optimism/op-service/crypto"
+	"github.com/ethereum-optimism/optimism/op-service/ctxinterrupt"
+	"github.com/ethereum-optimism/optimism/op-service/ioutil"
+	"github.com/ethereum-optimism/optimism/op-service/jsonutil"
+	oplog "github.com/ethereum-optimism/optimism/op-service/log"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/urfave/cli/v2"
+)
+
+type ImplementationsConfig struct {
+	L1RPCUrl                        string             `cli:"l1-rpc-url"`
+	PrivateKey                      string             `cli:"private-key"`
+	ArtifactsLocator                *artifacts.Locator `cli:"artifacts-locator"`
+	L1ContractsRelease              string             `cli:"l1-contracts-release"`
+	MIPSVersion                     int                `cli:"mips-version"`
+	WithdrawalDelaySeconds          uint64             `cli:"withdrawal-delay-seconds"`
+	MinProposalSizeBytes            uint64             `cli:"min-proposal-size-bytes"`
+	ChallengePeriodSeconds          uint64             `cli:"challenge-period-seconds"`
+	ProofMaturityDelaySeconds       uint64             `cli:"proof-maturity-delay-seconds"`
+	DisputeGameFinalityDelaySeconds uint64             `cli:"dispute-game-finality-delay-seconds"`
+	SuperchainConfigProxy           common.Address     `cli:"superchain-config-proxy"`
+	ProtocolVersionsProxy           common.Address     `cli:"protocol-versions-proxy"`
+	UseInterop                      bool               `cli:"use-interop"`
+
+	Logger log.Logger
+
+	privateKeyECDSA *ecdsa.PrivateKey
+}
+
+func (c *ImplementationsConfig) Check() error {
+	if c.L1RPCUrl == "" {
+		return errors.New("l1RPCUrl must be specified")
+	}
+	if c.PrivateKey == "" {
+		return errors.New("private key must be specified")
+	}
+
+	privECDSA, err := crypto.HexToECDSA(strings.TrimPrefix(c.PrivateKey, "0x"))
+	if err != nil {
+		return fmt.Errorf("failed to parse private key: %w", err)
+	}
+	c.privateKeyECDSA = privECDSA
+
+	if c.Logger == nil {
+		return errors.New("logger must be specified")
+	}
+	if c.ArtifactsLocator == nil {
+		return errors.New("artifacts locator must be specified")
+	}
+	if c.L1ContractsRelease == "" {
+		return errors.New("L1 contracts release must be specified")
+	}
+	if c.MIPSVersion != 1 && c.MIPSVersion != 2 {
```

Comment:
If `c.L1ContractsRelease == "dev"`, will this check be too restrictive? Same question for other checks throughout this function

Reply:
This is the MIPSVersion check - it allows any value of `L1ContractsRelease` that isn't empty. The MIPSVersion must be 1 or 2.